### PR TITLE
Pip: Add more VCS fallback URLs for packages

### LINKED ISSUE #509: PIP: Read `project_urls` attribute from `setup.py`
A VCS repository from which releases are created is preferrable to the location of a source artifact ("sdist" in Python lingo). There is a rarely used [attribute in `setup.py`](https://packaging.python.org/guides/distributing-packages-using-setuptools/#project-urls) where this information can be recorded. Make sure this information is captured, if it is present.

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add more fields that could contain a link to the source code repository
as fallback URLs for Python packages.

Fixes #509.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Pip.kt [hunk 1]
```diff
@@ -596,6 +596,15 @@ class Pip(
                 }
             }
 
+            val projectUrls = pkgInfo["project_urls"]
+            val vcsFallbackUrls = listOfNotNull(
+                homepageUrl,
+                pkgInfo["project_url"]?.textValue(),
+                projectUrls["Code"]?.textValue(),
+                projectUrls["Homepage"]?.textValue(),
+                projectUrls["Source"]?.textValue()
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Pip.kt:605]: @sschuberth In #509 you also mention "Source Code" as a possible field, I haven't seen this being used anywhere, shall I add it anyway?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Pip.kt:605]: Yes please. A quick search shows that "Source Code" is e.g. used at https://github.com/lepture/authlib/blob/c73e2a8921a15ee1f96b604123e49bd2d5fd651b/setup.cfg#L38 (via `setup.cfg`).

File: analyzer/src/main/kotlin/managers/Pip.kt [hunk 2]
```diff
@@ -596,6 +596,16 @@ class Pip(
                 }
             }
 
+            val projectUrls = pkgInfo["project_urls"]
+            val vcsFallbackUrls = listOfNotNull(
+                homepageUrl,
+                pkgInfo["project_url"]?.textValue(),
+                projectUrls["Code"]?.textValue(),
+                projectUrls["Homepage"]?.textValue(),
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pip.kt:604]: Nit: Does it make sense to swap the order here, to move `"Homepage"` closer to `homepageUrl` and `"Code"` closer to `"Source Code"`?

**Review by sschuberth — APPROVED** [review]

File: analyzer/src/main/kotlin/managers/Pip.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Pip.kt:604]: I'd rather keep this list alphabetically sorted.

File: analyzer/src/funTest/assets/projects/external/example-python-flask-expected-output-windows.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ project:
       dependencies:
       - id: "PyPI::click:8.1.3"
         dependencies:
-          - id: "PyPI::colorama:0.4.4"
+        - id: "PyPI::colorama:0.4.4"
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/external/example-python-flask-expected-output-windows.yml:25]: I guess this indentation was wrong already in the previous commit.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**